examples/gateway/proxy/blockstore.go

```diff
@@ -0,0 +1,111 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+
+	"github.com/ipfs/go-cid"
+	blockstore "github.com/ipfs/go-ipfs-blockstore"
+	"github.com/ipfs/go-libipfs/blocks"
+)
+
+var (
+	errNotImplemented = errors.New("not implemented")
+)
+
+type proxyStore struct {
+	httpClient *http.Client
+	host       string
+	validate   bool
+}
+
+func newProxyStore(host string, client *http.Client) blockstore.Blockstore {
+	if client == nil {
+		client = http.DefaultClient
+	}
+
+	return &proxyStore{
+		host:       host,
+		httpClient: client,
+	}
+}
+
+func (ps *proxyStore) fetch(ctx context.Context, c cid.Cid) (blocks.Block, error) {
+	u, err := url.Parse(fmt.Sprintf("http://%s/ipfs/%s?format=raw", ps.host, c))
+	if err != nil {
+		return nil, err
+	}
+	resp, err := ps.httpClient.Do(&http.Request{
+		Method: http.MethodGet,
+		URL:    u,
+		Header: http.Header{
+			"Accept": []string{"application/vnd.ipld.raw"},
+		},
+	})
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Need to check Status code of the response and return remote error if was anything other than 200.

For example, ipfs.io returns 500 and that breaks page:

> ![2023-02-03-021516_677x81_scrot](https://user-images.githubusercontent.com/157609/216487900-a168260d-2691-40bb-a3aa-cf0c22476a8b.png)

```console
$  curl 'https://ipfs.io/ipfs/filecoin.io?format=raw' | xxd 
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   194  100   194    0     0    710      0 --:--:-- --:--:-- --:--:--   710
00000000: 3c68 746d 6c3e 0d0a 3c68 6561 643e 3c74  <html>..<head><t
00000010: 6974 6c65 3e35 3033 2053 6572 7669 6365  itle>503 Service
00000020: 2054 656d 706f 7261 7269 6c79 2055 6e61   Temporarily Una
00000030: 7661 696c 6162 6c65 3c2f 7469 746c 653e  vailable</title>
00000040: 3c2f 6865 6164 3e0d 0a3c 626f 6479 3e0d  </head>..<body>.
00000050: 0a3c 6365 6e74 6572 3e3c 6831 3e35 3033  .<center><h1>503
00000060: 2053 6572 7669 6365 2054 656d 706f 7261   Service Tempora
00000070: 7269 6c79 2055 6e61 7661 696c 6162 6c65  rily Unavailable
00000080: 3c2f 6831 3e3c 2f63 656e 7465 723e 0d0a  </h1></center>..
00000090: 3c68 723e 3c63 656e 7465 723e 6f70 656e  <hr><center>open
000000a0: 7265 7374 793c 2f63 656e 7465 723e 0d0a  resty</center>..
000000b0: 3c2f 626f 6479 3e0d 0a3c 2f68 746d 6c3e  </body>..</html>
000000c0: 0d0a
```

Reply:
I made some updates, but I don't think we're gonna get exactly what you're expecting. Don't forget that this is just a block store and a routing mechanism and we have to conform to the interfaces. I can't just return the error so I build an error with that info.